fedmsg/api.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of fedmsg.
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+"""The main interface for fedmsg."""
+from __future__ import absolute_import
+import logging
+import json
+import threading
+
+import six
+import zmq
+
+from . import config, exceptions
+from .broker.service import SUBMISSION_V1
+
+_log = logging.getLogger(__name__)
+
+# Cache connections to the broker in a thread-local
+_thread_local = threading.local()
+
+
+def publish(topic=u'', headers=None, body=None, timeout=10):
```

Comment:
Do people think it's worth making this as drop-in replaceable for the old publish API as possible? The headers kwarg would need to be at the end and there would need to be some unused kwargs. It's not appealing because it will make it confusing to new users, but it would make transitioning easier, I suppose.

Reply:
No strong opinion on this.

Reply:
I would say keep it as API-compatible as possible and introduce a new API, maybe in an `api2` module?